src/utils/IMagicSpend.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
+
+/// @notice Minimal interface for MagicSpend (https://github.com/coinbase/magic-spend).
+interface IMagicSpend {
```

Comment:
I think we should prefer using the exact reference in `/lib`? Doesn't affect bytecode size

Reply:
Aligned. Was having foundry compilation errors so will fix this in a follow up PR this week. Want to get this merged for internal security review asap for now.